src/components/node-list/index.js

```diff
@@ -54,11 +58,26 @@ class NodeList extends React.Component {
 
   /**
    * Return only the results that match the search text
-   * @param {object} results
+   * @param {object} nodes
    */
-  filterResults(results) {
-    return results.filter(node =>
-      this.nodeMatchesSearch(node, this.state.searchValue)
+  filterResults(nodes) {
+    const obj = {};
+    Object.keys(nodes).forEach(key => {
+      obj[key] = nodes[key].filter(node =>
+        this.nodeMatchesSearch(node, this.state.searchValue)
+      );
+    });
+    return obj;
+  }
+
+  /**
+   * Get a list of IDs of the visible nodes
+   * @param {object} filteredNodes
+   */
+  getNodeIDs(filteredNodes) {
```

Comment:
This seems more of a helper / mapper function rather than needing to belong to the class itself, have you considered moving this out of the class? (would be good to have some test coverage of this as well)

Reply:
Yeah that's fair. I could move it out of the class, but it's basically only used on the class's state, and I don't think it'd be useful to abstract it at this point. Personally I think it doesn't really matter either way whether it's a class method or a standalone function, so I kept it as a method for neatness, so that it's kept with the other related functions. Do you disagree? If so I don't mind moving it.

The test coverage is a good shout. I'll add that

Reply:
Maybe it's something that can still live in this file (at the top) and can be exported to have some test coverage around it

Reply:
done